[25.05] matrix-authentication-service: 0.19.0 -> 0.20.0

teutat3s opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Manual backport of https://github.com/NixOS/nixpkgs/pull/431271 (and part of a [treewide](https://github.com/mdaniels5757/nixpkgs/commit/398b16e16cc688d8ffcf820d1ec50a55da44a108)) to `release-25.05`.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

teutat3s committed:
matrix-authentication-service: avoid updating to pre-releases

(cherry picked from commit 93f474f607a7774cadeaf9b1ee3dc8a8a043ce74)

Videl reviewed (approved):
Built, it works.

fpletz commented:
> Couldn't locate original commit hash in message of [3b2aee0](https://github.com/NixOS/nixpkgs/commit/3b2aee00c74ffaac4a8f10455aec065bd45fb98b).

This is IMHO partly justified since only a small fraction of the mentioned commit was backported. It does exist on master however, so the message is a bit misleading. I would prefer if we didn't mention the cherry pick in this case at all since the changes weren't backported completely.

Additionally, I'm unsure if we should wait to backport this because the whole commit is going to be backported anyway as mentioned in https://github.com/NixOS/nixpkgs/pull/421558#issuecomment-3121597471. /cc @niklaskorz

teutat3s committed:
matrix-authentication-service: 0.19.0 -> 0.20.0

Changelog: https://github.com/element-hq/matrix-authentication-service/releases/tag/v0.20.0
(cherry picked from commit 25fb6b71b2fe8f0bc6625504ced8af0ada6a8406)

teutat3s commented:
@fpletz I've removed the partial backport now.

fpletz reviewed (approved):
Thank you! Built on x86_64-linux and aarch64-linux.

niklaskorz commented:
> Additionally, I'm unsure if we should wait to backport this because the whole commit is going to be backported anyway as mentioned in [#421558 (comment)](https://github.com/NixOS/nixpkgs/pull/421558#issuecomment-3121597471). /cc @niklaskorz

Thanks for the reminder, as #423228 has reached release-25.05 now the backport of the treewide can start
